jadx-core/src/main/java/jadx/core/dex/visitors/rename/RenameVisitor.java

```diff
@@ -130,11 +130,12 @@ private static void checkPackage(Deobfuscator deobfuscator, ClassNode cls, Class
 	private static String fixClsShortName(JadxArgs args, String clsName) {
 		boolean renameValid = args.isRenameValid();
 		if (renameValid) {
-			char firstChar = clsName.charAt(0);
-			if (Character.isDigit(firstChar)) {
+			if (clsName.matches("^\\d+$")) {
```

Comment:
Please split compile and matches parts for better performance:
```java
// additional field
private static final Pattern ANONYMOUS_CLASS_PATTERN = Pattern.compile("^\\d+$");

// replace condition with
if (ANONYMOUS_CLASS_PATTERN.matcher(clsName).matches()) {
```

Reply:
Fixed it